src/wp-includes/blocks.php

```diff
@@ -883,16 +877,71 @@ function serialize_block( $block, $callback = null ) {
  * parsed blocks.
  *
  * @since 5.3.1
- * @since 6.4.0 The `$callback` parameter was added.
  *
- * @param array[]       $blocks   An array of representative arrays of parsed block objects. See serialize_block().
- * @param callable|null $callback Optional. Callback to run on each block in the tree before serialization. Default null.
+ * @param array[] $blocks An array of representative arrays of parsed block objects. See serialize_block().
+ * @return string String of rendered HTML.
+ */
+function serialize_blocks( $blocks ) {
+	return implode( '', array_map( 'serialize_block', $blocks ) );
+}
+
+/**
+ * Traverses the block applying transformations using the callback provided and returns the content of a block,
+ * including comment delimiters, serializing all attributes from the given parsed block.
+ *
+ * This should be used when there is a need to modify the saved block.
+ * Prefer `serialize_block` when preparing a block to be saved to post content.
+ *
+ * @since 6.4.0
+ *
+ * @see serialize_block()
+ *
+ * @param array         $block    A representative array of a single parsed block object. See WP_Block_Parser_Block.
+ * @param callable|null $callback Callback to run on each block in the tree before serialization.
+ * @return string String of rendered HTML.
+ */
+function traverse_and_serialize_block( $block, $callback ) {
+	$block = call_user_func( $callback, $block );
```

Comment:
I know we had it here for `serialize_block`, but I'd like to apply it to the [inner block passed to `traverse_and_serialize_block`](https://github.com/WordPress/wordpress-develop/pull/5246/files#diff-8c99af92e4ec0fdb307ddd9b42be1e1ef1efe4a9f31287c23f346244dddd1ce9R912) instead, as this will allow us to pass the parent, block index, and chunk index (see https://github.com/WordPress/wordpress-develop/pull/5242) 😄 

(As a consequence, this means we'll also have to apply the callback in `traverse_and_serialize_blocks`.)

Reply:
Sure, I saw that earlier today but I forgot about it.

Reply:
Done with https://github.com/WordPress/wordpress-develop/pull/5246/commits/d008f9c050a9da8bac463e7a7073248d72e36d55.